storage/interceptors/smaap_service_instance_interceptor.go

```diff
@@ -447,7 +444,31 @@ func preparePrerequisites(ctx context.Context, repository storage.Repository, os
 	return osbClient, broker, service, plan, nil
 }
 
-func (i *ServiceInstanceInterceptor) prepareProvisionRequest(instance *types.ServiceInstance, serviceCatalogID, planCatalogID string) *osbc.ProvisionRequest {
+func (i *ServiceInstanceInterceptor) prepareProvisionRequest(instance *types.ServiceInstance, serviceCatalogID, planCatalogID string) (*osbc.ProvisionRequest, error) {
+	context := make(map[string]interface{})
+	if len(instance.Context) != 0 {
+		if err := json.Unmarshal(instance.Context, &context); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal already present OSB context: %s", err)
+		}
+	} else {
+		context = map[string]interface{}{
+			"platform":      types.SMPlatform,
+			"instance_name": instance.Name,
+		}
+
+		if len(i.tenantKey) != 0 {
+			if tenantValue, ok := instance.GetLabels()[i.tenantKey]; ok {
+				context[i.tenantKey] = tenantValue[0]
+			}
+		}
+
+		contextBytes, err := json.Marshal(context)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal OSB context %+v: %s", context, err)
+		}
+		instance.Context = contextBytes
```

Comment:
This is a bit hidden side effect. Can we ensure the instance has a context in another function?